File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        int N = Integer.parseInt(br.readLine());
        int T = Integer.parseInt(br.readLine());
        Queue<Integer> que = new LinkedList<>();
        List<List<Integer>> list = new ArrayList<>();
        boolean[] ischecked = new boolean[N + 1];
        for(int i = 0; i <= N; i++) {
            list.add(new ArrayList<>());
        }
        for(int tc = 0; tc < T; tc++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            list.get(a).add(b);
            list.get(b).add(a);
        }
        que.add(1);
        ischecked[1] = true;
        int cnt = 0;
        while(!que.isEmpty()) {
            int tmp = que.poll();   
            for(int num : list.get(tmp)) {
                if(!ischecked[num]) {
                    que.add(num);
                    ischecked[num] = true;
                    cnt++;
                }                
            }
        }
        System.out.println(cnt);
    }
}

File: 백준/Silver/2606. 바이러스/바이러스.js
class Queue {
  constructor() {
    this.arr = [];
    this.head = 0;
  }

  add(num) {
    this.arr.push(num);
  }

  poll() {
    return this.arr[this.head++];
  }

  isEmpty() {
    return this.head === this.arr.length;
  }
}

let [N, T, ...input] = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n').map(line => line.trim());
N = Number(N);
T = Number(T);
input = input.map(line => line.split(' ').map(Number));
arr = Array.from({length : N + 1}, () => []);
for(let i = 0; i < T; i++) {
  arr[input[i][0]].push(input[i][1]);
  arr[input[i][1]].push(input[i][0]);
}
const que = new Queue();
const visited = new Array(N + 1).fill(false);
que.add(1);
visited[1] = true;
let cnt = 0;
while(!que.isEmpty()) {
  const num = que.poll();
  for(let tmp of arr[num]) {
    if(!visited[tmp]) {
      que.add(tmp);
      visited[tmp] = true;
      cnt++;
    }
  }
}
console.log(cnt);